src/expr/src/explain/text.rs

```diff
@@ -362,17 +364,70 @@ impl MirRelationExpr {
                     })?;
                 }
             }
-            Get { id, .. } => {
+            Get {
+                id,
+                persist_or_index,
+                ..
+            } => {
                 match id {
                     Id::Local(id) => {
+                        assert!(persist_or_index.is_none());
                         write!(f, "{}Get {}", ctx.indent, id)?;
                     }
                     Id::Global(id) => {
-                        let humanized_id = ctx
-                            .humanizer
-                            .humanize_id(*id)
-                            .unwrap_or_else(|| id.to_string());
-                        write!(f, "{}Get {}", ctx.indent, humanized_id)?;
+                        let humanize = |id: &GlobalId| {
+                            ctx.humanizer
+                                .humanize_id(*id)
+                                .unwrap_or_else(|| id.to_string())
+                        };
+                        let humanize_unqualified = |id: &GlobalId| {
+                            ctx.humanizer
+                                .humanize_id_unqualified(*id)
+                                .unwrap_or_else(|| id.to_string())
+                        };
+                        let humanize_unqualified_maybe_deleted = |id: &GlobalId| {
+                            ctx.humanizer
+                                .humanize_id_unqualified(*id)
+                                .unwrap_or("[DELETED INDEX]".to_owned())
+                        };
+                        match persist_or_index {
+                            None => {
+                                write!(f, "{}Get {}", ctx.indent, humanize(id))?;
+                            }
+                            Some(PersistOrIndex::Persist) => {
+                                write!(f, "{}ReadStorage {}", ctx.indent, humanize(id))?;
```

Comment:
I was thinking of `ReadIndex` / `ReadStorage` as virtual nodes. Note that `ReadIndex` captures not just certain Gets, but also the `IndexedFilter` pattern, see `fmt_indexed_filter` (called both from fast path and dataflow). So, if I modify the `Get` printing of the `ReadIndex` case to be `Get $name access=index ...`, then should I also modify `fmt_indexed_filter` to print `Get $name access=index ...`, or should that one remain `ReadIndex`? If I modify it to be `Get $name access=index ...`, then it's weird, because it's not really just a `Get`. But if I keep it `ReadIndex`, then it might be confusing for readers what is the difference between `ReadIndex` and `Get $name access=index ...`. So this is why I thought that a virtual node `ReadIndex` capturing both cases is the best option.

For the syntax, I didn't follow the MIR EXPLAIN AS TEXT guidelines because I thought that's only important if the MIR parser will have to parse it, which is not the case here, because we can just keep using `Get` in the spec tests. So I tried to optimize it for human readability. But I'm open to refining the syntax. Your syntax proposal also looks well-readable. I guess we should first decide on whether to have the virtual nodes, and then figure out the syntax details.

Reply:
> I was thinking of ReadIndex / ReadStorage as virtual nodes.

Understood. In that case, let's stick with the virtual syntax! As follow ups, we should:

1. Ensure that `EXPLAIN WITH(raw_syntax)` reverts to not printing this.
2. Check if we can also skip the enclosing `ArrangeBy` if it refers to one of the indexes printed by the virtual node.

> So I tried to optimize it for human readability.

I see. The syntax guidelines predate the MIR parser (I'm not too worried about that) and were also trying to optimize for human readability. My idea was that if all nodes at the same IR level (MIR, LIR) are rendered using a consistent set of rules it will be easier to train your eyes when looking for a specific information. Also, searching within a string buffer `ctrl+f` also becomes a bit more predictable.

Reply:
Ok, so I will stay with the virtual nodes.

But I will change the syntax to what you wrote:
```
ReadIndex on=$humanized_id $idx_name_A=[$usage_types_A] $idx_name_B=[$usage_types_B] ...
```

> Ensure that EXPLAIN WITH(raw_syntax) reverts to not printing this.

I noted this down in my todos. Will do hopefully the week after the onsite.

> Check if we can also skip the enclosing ArrangeBy if it refers to one of the indexes printed by the virtual node.

Yes, I'll do this at some point. But even before that, I'm planning to start a discussion in the compute slack channel about renaming MIR `ArrangeBy` to something like `ArrangementNeeded`, which is hopefully a little less misleading:  `ArrangeBy` sounds totally like we are creating an arrangement right there, while with `ArrangementNeeded` the user will at least pause for a moment and think about what this might be and will hopefully either check the docs or ask us instead of simply misunderstanding it.

Reply:
Done